x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -415,6 +417,11 @@ private static void checkEvalFields(LogicalPlan p, Set<Failure> failures) {
                     MatchQueryPredicate.class,
                     mqp -> { failures.add(fail(mqp, "EVAL does not support MATCH expressions")); }
                 );
+                // check no full text search functions are used
+                field.forEachDown(
```

Comment:
Can this be moved into `checkFullTextQueryFunctions` so that there is one place to validate FullTextQuery related restrictions?